File: student/src/main/java/com/ktxdevelopment/model/entity/Student.java
package com.ktxdevelopment.model.entity

import jakarta.persistence.*
import org.springframework.context.annotation.Lazy;


@Table(name = "students")
@Entity(name = "student")
class Student {

    @Id
    @GeneratedValue
    private Integer id;

    @Column(unique = true, name = "student_id")
    private String studentId;

    @Column(unique = false, name = "first_name")
    private String firstName;

    @Column(unique = false, name = "last_name")
    private String lastName;

    @Column(unique = false, name = "profile_image")
    private String profileImage;


    private Course course;

    @OneToOne
    @Lazy
    private Transcript transcript;

    @OneToOne
    private PaymentProfile paymentProfile;
}

File: student/src/main/java/com/ktxdevelopment/model/entity/Student.kt
package com.ktxdevelopment.model.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Table
import org.springframework.data.annotation.Id


@Entity
@Table
data class Student(
    @Id
    @GeneratedValue
    val id: Long,

    @Column(unique = false,name = "first_name")
    val firstName: String,
    @Column(unique = false, name = "Last_name")
    val lastName: String,
    @Column(unique = false, name = "Last_name")
    val profileImage: String,
    val entryYear: Int,
    val gpa: Int = 0,
    val faculty: Faculty,
    val courses: List<Course> = listOf()
)